pages/api/nft/loan/[positionId].ts

```diff
@@ -0,0 +1,107 @@
+import { promises as fs } from 'fs'
+
+import type { NextApiRequest, NextApiResponse } from 'next'
+import BigNumber from 'bignumber.js'
+
+import { getBuiltGraphSDK } from '../../../../.graphclient'
+import { fetchPrices } from '../../../../apis/currency'
+import { toCurrency } from '../../../../apis/asset'
+import { dollarValue, formatUnits } from '../../../../utils/numbers'
+
+const { getLoanPosition } = getBuiltGraphSDK()
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<any>,
+) {
+  try {
+    const query = req.query
+    const { positionId } = query
+    if (!positionId || typeof positionId !== 'string') {
+      res.json({
+        status: 'error',
+        message: 'Something went wrong, please try again!!!',
+      })
+      return
+    }
+    const { loanPosition } = await getLoanPosition(
+      {
+        positionId: positionId,
+      },
+      {
+        url: process.env.SUBGRAPH_URL,
+      },
+    )
+    if (!loanPosition) {
+      res.json({
+        status: 'error',
+        message: 'Something went wrong, please try again!!!',
+      })
+      return
+    }
+    const loanAmount = BigInt(loanPosition.amount)
+    const collateralAmount = BigInt(loanPosition.collateralAmount)
+    const loanToken = toCurrency(loanPosition.underlying)
```

Comment:
Not token, use `Currency`
```suggestion
    const debtCurrency = toCurrency(loanPosition.underlying)
```